crates/core/src/fork.rs

```diff
@@ -548,6 +548,7 @@ impl ForkDetails {
             )
         })?;
         let l1_batch_number = block_details.l1_batch_number;
+        block.l1_batch_number = Some(l1_batch_number.0.into());
```

Comment:
TBH I have feeling this might lead to some unexpected behaviour because that block might technically not be sealed in that batch yet (this is what `l1BatchNumber == null` means). But anyway, the whole forking logic needs some rethinking in general so I am fine with keeping as is